build.zig

```diff
@@ -156,6 +166,19 @@ pub fn build(b: *zbs.Builder) !void {
     }
 }
 
+fn writePkgconfigFile(writer: anytype, prefix: []const u8) !void {
+    try writer.print(
+        \\prefix={s}
+        \\datarootdir=${{prefix}}/share
+        \\pkgdatadir=${{datarootdir}}/river-protocols
```

Comment:
the wayland protocols pkg config file does
```
pkgdatadir=${pc_sysrootdir}${datarootdir}/wayland-protocols
```
What's the difference here?

Reply:
Wayland protocols use some variables from meson see https://gitlab.freedesktop.org/wayland/wayland-protocols/-/blob/main/meson.build#L88 and also see this commit message https://gitlab.freedesktop.org/wayland/wayland-protocols/-/commit/1ad48a50a45b4a4dbc84d9e85ea49311bb864629

So to be honest  I'm not sure if we need an hack like this or not, pkg-config documentation use variable like I did

Reply:
I was more talking about the `pc_sysrootdir` present in the wayland protocols .pc file but not here. If I understand correctly, that's required for `PKG_CONFIG_SYSROOT_DIR` to work. It seems to me that wayland-protocols does this wrong though and it should be used to define the `prefix` variable with
```
prefix=${{pc_sysrootdir}}/{s}
```

We don't need the `datarootdir` hack from wayland-protocols as we don't and never will use autotools, let's switch to the more standard `datadir`.

Reply:
Hmm yeah you're right I misunderstood that, so do we use
```
        \\prefix=${{pc_sysrootdir}}/{s}
        \\datadir=${{prefix}}/share
        \\pkgdatadir=${{datadir}}/river-protocols
```
or  just
```
        \\prefix=${{pc_sysrootdir}}/{s}
        \\datadir=${{prefix}}/share/river-protocols
```

Reply:
You know what, grepping around my system almost nothing seems to use `pc_sysrootdir`. Let's just leave it out unless someone complains. I think this should be good:
```
        \\prefix={s}
        \\datadir=${{prefix}}/share
        \\pkgdatadir=${{datadir}}/river-protocols

```

Reply:
Same, only python use it IIRC